shoop/core/models/_service_payment.py

```diff
@@ -135,6 +138,16 @@ def get_service_choices(self):
             ServiceChoice('cash', _("Cash payment"))
         ]
 
+    def _create_service(self, choice_identifier, **kwargs):
+        service = super(CustomPaymentProcessor, self)._create_service(
+            choice_identifier, **kwargs)
+        if choice_identifier == 'cash':
+            service.behavior_components.add(
+                RoundingBehaviorComponent.objects.create())
+            service.behavior_components.add(
+                StaffOnlyBehaviorComponent.objects.create())
+        return service
```

Comment:
Yep, just what I meant. Good work!

Btw. it's possible to add several objects in one call, like `service.behavior_components.add(comp1, comp2)`.